### PR TITLE
[ENG-475] Meeting detail header

### PR DESCRIPTION
**Pull request by adlius**
## Purpose

Add meeting detail page header.

## Summary of Changes

New `meeting-detail-header` component and some tests.

## Ticket

https://openscience.atlassian.net/browse/ENG-475

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: app/meetings/detail/-components/meeting-detail-header/template.hbs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                <li>{{this.typeOneEmailLineText}}</li>
+            {{/if}}
+            
+            {{#if this.meeting.isAcceptingTypeTwo}}
+                <li>{{this.typeTwoEmailLineText}}</li>
+            {{/if}}
+        </ul>
+        <p data-test-meeting-format-header>{{t 'meetings.detail.meeting-detail-header.format_header'}}</p>
+        <div data-test-meeting-format-body>
+            <dl style='padding-left: 25px'>
```
Comment (jamescdavis) [inline app/meetings/detail/-components/meeting-detail-header/template.hbs]: Please move this style into a local-class.

File: app/locales/en/translations.ts [hunk 2]
```diff
@@ -1259,6 +1259,28 @@ export default {
...
+                category: 'Category',
+                dateCreated: 'Date Created',
+                downloads: 'Downloads',
+                noSubmissions: 'No Submissions',
+            },
+            'meeting-detail-header': {
+                image_alt: 'Meeting logo image',
+                add_your: 'Add your',
+                send_email: 'Send an email to the following address(es) from the email account you would like used on the OSF:',
+                email_text: 'For {{typeName}}, email <a href="{{emailAddress}}">{{emailAddress}}</a>',
```
[the remark this answers is not in the document]
Comment (adlius) [reply app/locales/en/translations.ts]: We seem to have an established paradigm to put email link text in the translation strings. To name a few:
https://github.com/CenterForOpenScience/ember-osf-web/blob/develop/app/locales/en/translations.ts#L350
https://github.com/CenterForOpenScience/ember-osf-web/blob/develop/app/locales/en/translations.ts#L360
https://github.com/CenterForOpenScience/ember-osf-web/blob/develop/app/locales/en/translations.ts#L601

File: app/meetings/detail/-components/meeting-detail-header/component.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        });
+    }
+
+    @computed('meeting.fieldNames')
+    get conferenceHomepageText() {
+        if (this.meeting.fieldNames.homepage_link_text) {
+            return this.meeting.fieldNames.homepage_link_text;
+        }
+        return this.i18n.t('meetings.detail.meeting-detail-header.conference_homepage');
+    }
```
[the remark this answers is not in the document]
Comment (adlius) [reply app/meetings/detail/-components/meeting-detail-header/component.ts]: The text depends on a lot of stuff. The reason I used computed properties is to make the template cleaner. I personally believe that tracing different conditions in JS/TS is much easier than tracing different conditions in handlebar.

[the remark this answers is not in the document]
Comment (jamescdavis) [reply app/meetings/detail/-components/meeting-detail-header/component.ts]: Ok, that's fair, especially for `panelToggleLinkText` and `panelHeaderText`, but the `addYour` is not dependent on conditions and should just be in the template. I'd like to see `panelToggleLinkText` and `panelHeaderText` return *just* `addSubmission`/`submissionOne`/`submissionTwo` and in the template have, e.g.:
```hbs
{{t 'meetings.detail.meeting-detail-header.add_your'} {{this.panelToggleLinkText}}
```

`typeOneEmailLineText` and `typeTwoEmailLineText` have no conditions so, IMHO, should just be in the template, e.g.
```hbs
{{t 'meetings.detail.meeting-detail-header.email_text'
    typeName=this.meeting.fieldNames.submission1_plural
    emailAddress=this.meeting.typeOneSubmissionEmail
}}
```

Also, `conferenceHomepageText` is a simple condition that can expressed as:
```hbs
{{if this.meeting.fieldNames.homepage_link_text
    this.meeting.fieldNames.homepage_link_text
    (t 'meetings.detail.meeting-detail-header.conference_homepage')
}}
```
which is pretty easy to read IMHO.

File: app/meetings/detail/-components/meeting-detail-header/component.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+        return dateString;
+    }
+
+    @computed('meeting.{fieldNames,isAcceptingTypeOne,isAcceptingTypeTwo}')
+    get panelToggleLinkText() {
+        const { isAcceptingTypeOne, isAcceptingTypeTwo } = this.meeting;
+        const addSubmission = this.meeting.fieldNames.add_submission;
+        const submissionOne = this.meeting.fieldNames.submission1;
+        const submissionTwo = this.meeting.fieldNames.submission2;
```
Comment (jamescdavis) [inline app/meetings/detail/-components/meeting-detail-header/component.ts]: These should also be plural, see: https://test.osf.io/view/opensyllabi/

File: app/meetings/detail/-components/meeting-detail-header/template.hbs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+{{#if this.meeting.logoUrl}}
+    <div>
+        <img data-test-meeting-logo src={{this.meeting.logoUrl}} alt={{t 'meetings.detail.meeting-detail-header.image_alt'}} class='img-responsive'>
+    </div>
+{{/if}}
+
+{{#if this.meeting.active}}
+    <div local-class='toggle-panel-button'>
+        <OsfButton data-analytics-name='Toggle meeting submission panel' data-test-meeting-toggle-panel-button @type='link' @onClick={{action this.togglePanel}}>
```
Comment (jamescdavis) [inline app/meetings/detail/-components/meeting-detail-header/template.hbs]: Please break this into multiple lines. A rule I've been using is break into multiple line if there are >1 attributes.

File: app/meetings/detail/-components/meeting-detail-header/component.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    get locationAndDateText() {
+        if (!this.meeting) {
+            return null;
+        }
+        const { startDate, endDate, location } = this.meeting;
+        let dateString = null;
+        if (startDate && endDate) {
+            dateString = `${moment(startDate).format('MMMM Do, YYYY')} - ${moment(endDate).format('MMMM Do, YYYY')}`;
+        } else {
+            dateString = `${moment(startDate).format('MMMM Do YYYY')}`;
```
Comment (jamescdavis) [inline app/meetings/detail/-components/meeting-detail-header/component.ts]:
```suggestion
            dateString = `${moment(startDate).format('MMMM Do, YYYY')}`;
```

File: app/meetings/detail/-components/meeting-detail-header/template.hbs [hunk 7]
```diff
@@ -0,0 +1,73 @@
+<h1 data-test-meeting-name>{{this.meeting.name}}</h1>
+
+<div data-test-meeting-location-and-date local-class='location-and-date-text'>
+    {{this.locationAndDateText}}
+</div>
+
+{{#if this.meeting.logoUrl}}
+    <div>
+        <img data-test-meeting-logo src={{this.meeting.logoUrl}} alt={{t 'meetings.detail.meeting-detail-header.image_alt'}} class='img-responsive'>
```
Comment (jamescdavis) [inline app/meetings/detail/-components/meeting-detail-header/template.hbs]: Please break into multiple lines.

File: app/meetings/detail/-components/meeting-detail-header/template.hbs [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        {{if this.meeting.fieldNames.homepage_link_text 
+            this.meeting.fieldNames.homepage_link_text 
+            (t 'meetings.detail.meeting-detail-header.conference_homepage')
+        }}
+    </OsfLink>
+{{/if}}
+
+<CpPanel @open={{this.isPanelOpen}} as |panel|>
+    <panel.body>
+        <h2 data-test-meeting-panel-header local-class='panel-header-text'>{{t 'meetings.detail.meeting-detail-header.add_your'}} {{this.panelToggleLinkText}}</h2>
```
Comment (jamescdavis) [inline app/meetings/detail/-components/meeting-detail-header/template.hbs]:
```suggestion
        <h2 data-test-meeting-panel-header local-class='panel-header-text'>
            {{t 'meetings.detail.meeting-detail-header.add_your'}} {{this.panelToggleLinkText}}
        </h2>
```

File: app/meetings/detail/-components/meeting-detail-header/component.ts [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+        if (location) {
+            return `${location} | ${dateString}`;
+        }
+
+        return dateString;
+    }
+
+    @computed('meeting.{fieldNames,isAcceptingTypeOne,isAcceptingTypeTwo}')
+    get panelToggleLinkText() {
```
Comment (jamescdavis) [inline app/meetings/detail/-components/meeting-detail-header/component.ts]:
```suggestion
    get addSubmissionText() {
```

File: app/meetings/detail/-components/meeting-detail-header/component.ts [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        } else {
+            dateString = `${moment(startDate).format('MMMM Do YYYY')}`;
+        }
+
+        if (location) {
+            return `${location} | ${dateString}`;
+        }
+
+        return dateString;
+    }
```
Comment (jamescdavis) [inline app/meetings/detail/-components/meeting-detail-header/component.ts]: IMHO, this is nicer in the template as:
```hbs
{{if this.meeting.location (concat this.meeting.location ' | ')}}
{{moment-format this.meeting.startDate 'MMMM Do, YYYY'}}
{{if this.meeting.endDate (concat ' - ' (moment-format this.meeting.endDate 'MMMM Do, YYYY')}}
```

Comment (jamescdavis) [reply app/meetings/detail/-components/meeting-detail-header/component.ts]: Also, if we're matching the existing page, the date format should be `MMM DD, YYYY` (see: https://test.osf.io/view/SGGL2018/).

**Comment by jamescdavis** [comment]
Also, please update with the latest `develop`.

**Pull request merged**